test/extended/util/framework.go

```diff
@@ -236,127 +236,152 @@ func processScanError(log string) error {
-							return processScanError(fmt.Sprintf("after seeing FailedImageImports for %s a manual image-import was submitted", lang))
+			return false, err
+		}
+		if retried {
+			return false, nil
+		}
+		return checkOpenShiftNamespaceImageStreamImported(ctx, oc, imagestream, registryHostname)
+	})
+	// pollErr will be not nil if there was an immediate error, or we timed out.
+	if pollErr == nil {
+		return nil
+	}
+	DumpImageStream(oc, "openshift", imagestream)
+	// If retried=true at this point, it means that we have repeatedly tried to reimport the imagestream and failed to do so.
+	// This could be an indicator that the Red Hat Container Registry (registry.redhat.io) is experiencing an outage, since most samples operator imagestream images are hosted there.
+	if retried {
+		strbuf := bytes.Buffer{}
+		strbuf.WriteString("Failed immagestream imports may indicate an issue with the Red Hat Container Registry (registry.redhat.io).\n")
+		strbuf.WriteString(" - check status at https://status.redhat.com (catalog.redhat.com) for reported outages\n")
+		strbuf.WriteString(" - if no outages are reported there, email Terms-Based-Registry-Team@redhat.com with a report of the error\n")
+		strbuf.WriteString("   and prepare to work with the test platform team to get the current set of tokens for CI\n")
+		e2e.Logf(strbuf.String())
+	}
+	return pollErr
+}
+
+// retrySamplesOperatorImagestreamImport immediately retries an import for the provided imagestream if an imagestream managed by the samples operator fails to import.
+// This allows imagestreams to be reimported at a faster cadence than what the samples operator currently provides.
+// Imagestreams which use images in the OCP payload are not managed by the samples operator and therefore will not be retried.
+// Returns true if the imagestream import was retried.
+func retrySamplesOperatorImagestreamImport(ctx context.Context, oc *CLI, imagestream string) (bool, error) {
+	// check the samples operator to see about imagestream import status
+	samplesOperatorConfig, err := oc.AdminConfigClient().ConfigV1().ClusterOperators().Get(ctx, "openshift-samples", metav1.GetOptions{})
+	if err != nil {
+		return false, processScanError(fmt.Sprintf("failed to get clusteroperator for samples-operator: %v", err))
+	}
+	for _, condition := range samplesOperatorConfig.Status.Conditions {
+		switch {
+		case condition.Type == configv1.OperatorDegraded && condition.Status == configv1.ConditionTrue:
+			// if degraded, bail ... unexpected results can ensue
+			return false, processScanError(fmt.Sprintf("samples-operator is degraded with reason: %s", condition.Reason))
+		case condition.Type == configv1.OperatorProgressing:
+			// if the imagestreams for one of our langs above failed, we abort,
+			// but if it is for say only EAP streams, we allow
+			if condition.Reason == "FailedImageImports" {
+				msg := condition.Message
+				if strings.Contains(msg, " "+imagestream+" ") || strings.HasSuffix(msg, " "+imagestream) {
```

Comment:
@dperaza4dustbit - I actually made a mistake when talking with @adambkaplan about this section of the code, thinking it had to be redone.  It does not.

Quiz for you:  can you post some links from the code in https://github.com/openshift/cluster-samples-operator as to why this is correct, even after the rewrite I did in 4.7, where we started employing configmaps to track imagestream status?

Reply:
@dperaza4dustbit and I confirmed his understanding in slack ... and fwiw it basically correlates with my answer to @sthaha with https://github.com/openshift/origin/pull/26472/files#r715648199